src/units/abilities.cpp

```diff
@@ -1963,6 +1965,13 @@ effect::effect(const unit_ability_list& list, int def, const_attack_ptr att, boo
 			}
 		}
 
+		if((type != "illuminates") && cfg.has_attribute("max_value")){
+			max_value = max_value ? std::max(*max_value, cfg["max_value"].to_int()) : cfg["max_value"].to_int();
+		}
+		if((type != "illuminates") && cfg.has_attribute("min_value")){
+			min_value = min_value ? std::min(*min_value, cfg["min_value"].to_int()) : cfg["min_value"].to_int();
+		}
```

Comment:
This should be one outer if-statement for `type != "illuminates"`, and then checking for `max_value` and `min_value` done separately inside it.

Reply:
Also, what scenario would this be needed for? There's already a min/max check for illuminates in `tod_manager.cpp` isn't there?

Reply:
I thought about allowing value limitations in abilities like leadership or damage but I realized that the simultaneous use of 2 chance_to_hit or 2 leadership, one with max_value and the other with cumulative=yes risks interfere with the operation of the latter; so I plan to close this PR and add min_value only to the resistance abilities

Reply:
I'm not sure how that relates to my above question, but I have always wondered why only leadership is apparently affected by cumulative.

Reply:
Cumulative does not work the same way for leadership as it does for other abilities:

in leadership when cumulative=yes then each time and that several examples of the ability influence the same unit, the value will be the sum of each value=.

In special weapons like chance_to_hit, if cumulative=yes then the value used will be the highest between the base value and the value= of the special

Reply:
reopen this PR because someone add suggest in forum the idea of 2. Inacurrate
This would be an anti marksman, which allows for no more than 20% chance to hit in https://forums.wesnoth.org/viewtopic.php?p=689486#p689486 and i can be used with max_value and value=0 cumulative=yes

and another example would be a semi_magic weapon with a minimum like marksman (value=50) but also a maximum (70), so instead of an imposed fixed value, we would have a fairly easy to define value range, the reverse of the medal and that if a weapon adds a 10% chance of hitting to the whole, the maximum limit of 70% could not be exceeded in this case, but this kind of limitation could be precisely easier to encode than using [ filter_vase_value] and determining the base value as was the case in the Utbs 'training' ability before the quenoth lineage rework.